setup.sh

```diff
@@ -0,0 +1,382 @@
+write() {
+  echo -e " $1$2" "$NORMAL" >&2
+}
+
+writeBold() {
+  echo -e "${BOLD} $1$2" "$NORMAL" >&2
+}
+
+promptBold() {
+  echo -n -e "${BOLD} $1$2" "$NORMAL" >&2
+}
+
+commandExists() {
+  command -v "$@" > /dev/null 2>&1
+}
+
+checkRoot() {
+  if [ "$EUID" -ne 0 ]; then
+    echo
+    writeBold "$YELLOW" "[✖] This script needs to be executed with root privileges."
+    echo
+    exit 1
+  fi
+}
+
+findOSType()
+{
+  osType=$(uname)
+  if [ "$osType" == "Linux" ]; then
+    if [ -f /etc/os-release ]; then
+      distro=$(awk -F= '/^NAME/{print $2}' /etc/os-release)
+    else
+      distro=$(ls -d /etc/[A-Za-z]*[_-][rv]e[lr]* | grep -v "lsb" | cut -d'/' -f3 | cut -d'-' -f1 | cut -d'_' -f1)
+    fi
+    CURRENT_OS=$(echo $distro | tr 'a-z' 'A-Z' | cut -d' ' -f1 | tr -d '"')
+  fi
+}
+
+isOSSupported()
+{
+  echo
+  # echo "Detected OS is" $CURRENT_OS
+  case "$CURRENT_OS" in
+    "UBUNTU" | "DEBIAN")
+    {
+      write "$GREEN" "[✔] Your OS ($CURRENT_OS) is officially supported."
+      OS_IS_SUPPORTED=1
+    } ;;
+    *)
+    {
+      write "$YELLOW" "[✖] Your OS ($CURRENT_OS) is not officially supported."
+      write           "    This means we didn't thoroughly test Kuzzle on your"
+      write           "    system. It is likely to work, so you may want to continue."
+      OS_IS_SUPPORTED=0
+      while [[ "$proceedNotSupported" != [yYnN] ]]
+      do
+        promptBold "[❓] Do you want to continue? (y/N)"
+        read proceedNotSupported trash
+        case "$proceedNotSupported" in
+          [yY])
+            echo
+            writeBold "$GREEN" "Great! Join us on Gitter if you need help (https://gitter.im/kuzzleio/kuzzle)"
+            ;;
+          [nN] | '')
+            echo
+            writeBold "$BLUE" "Ok. We encourage you to get in touch with the team (tech@kuzzle.io)"
+            writeBold "$BLUE" "to request support for your system."
+            echo
+            if commandExists curl; then
+              # TODO send feedback to analytics system to notify that an install
+              # has been attempted on a non-supported system
+              echo LOL
+            fi
+            exit 2
+            ;;
+          *)
+            echo
+            writeBold "$RED" "[✖] I did not understand your answer."
+            ;;
+        esac
+      done
+    } ;;
+  esac
+}
+
+installDocker() {
+  echo
+  writeBold "[ℹ] Docker must be installed to run Kuzzle."
+  writeBold "    This script can install Docker for you, otherwise you can do it manually."
+  write     "    More information at https://docs.docker.com/engine/installation/"
+  while [[ "$installDocker" != [yYnN] ]]
+  do
+    promptBold "[❓] Do you want to install Docker now? (y/N)"
+    read installDocker trash
+    case $installDocker in
+      [yY])
+        echo
+        if commandExists curl; then
+          writeBold "[ℹ] Installing Docker..."
+          curl -sSL https://get.docker.com/ | sh
```

Comment:
Since `curl` is used multiple times in the script, I would add a prereq check to ensure it is installed. And if not, offer to the user to install it, explaining why it's necessary. 
If the user refuse, then:

* auto-abort the script if either docker or docker-compose is missing
* auto-refuse statistics upload

I've mixed feelings about `wget`: I agree it's a good fallback solution for docker and compose install, but we won't be able to use it to upload statistics. 
So... I see 2 ways of handling this case:

* if curl is absent but wget is there, and if the user refuse to install curl, then auto-refuse statistics upload but keep the option to install docker or docker-compsoe
* stick to curl only and forget about using wget